README.md

```diff
@@ -15,218 +15,380 @@ and understand how to run and work with Dgraph.
     - [Setting Metadata Headers](#setting-metadata-headers)
-    - [Connecting To Dgraph Cloud Endpoint](#connecting-to-dgraph-cloud-endpoint)
-    - [Cleanup Resources](#cleanup-resources)
 
 ## Install
 
 Install using nuget:
 
 ```sh
 dotnet add package Dgraph
-# or
-dotnet add package Dgraph --version 21.3.1.2
 ```
 
 >WARNING: Be aware that there may be other .NET packages with similar names. To verify the official package, please visit https://www.nuget.org/packages/Dgraph. Make sure you are using the correct and official package to avoid potential confusion.
 
 
-## Supported Versions
+## Supported 
 
-Each release of this client will support the equivalent Dgraph release. For example, 2020.03.XX will support any Dgraph instances with version 2020.03.XX. 
+Depending on the version of Dgraph that you are connecting to, you will have to use a different version of this client.
 
+|  Dgraph version  |  Dgraph.net version  |
+| ---------------- | -------------------- |
+|  dgraph 21.X.Y   |  Dgraph.net 21.3.1.2 |
+|  dgraph 23.X.Y   |  Dgraph.net 23.0.0   |
 
 ## Using a Client
 
 ### Creating a Client
 
-Make a new client by passing in one or more GRPC channels pointing to alphas.
+An `IDgraphClient` can be created with a list of `GrpcChannel` as objects. Connecting to multiple Dgraph servers in the same cluster allows for better distribution of workload.
+
+The following code snippet shows just one connection.
 
 ```c#
-var uri = new Uri("http://127.0.0.1:9080");
+using Dgraph;
+using Grpc.Net.Client;
+
+GrpcChannel channel = GrpcChannel.ForAddress("http://localhost:9080");
+using var dgraphClient = DgraphClient.Create(channel);
+```
+
+The connection can be configured by using `GrpcChannelOptions`.
+
+```c#
+using Grpc.Net.Client;
+
 var options = new GrpcChannelOptions
 {
-    Credentials = ChannelCredentials.Insecure
+    CompressionProviders = <...>, // List of Grpc ICompressionProvider
+    Credentials = Grpc.Core.ChannelCredentials.Create(<...>)
 };
-var client = new DgraphClient(GrpcChannel.ForAddress(uri, options));
+GrpcChannel channel = GrpcChannel.ForAddress("http://localhost:9080", options);
+```
+
+### Login into a namespace
+
+If your server has Access Control Lists enabled (Dgraph v1.1 or above), the client must be logged in for accessing data. Use `Login` to obtain and remember access and refresh JWTs.
+
+```c#
+var response = await dgraphClient.Login("user", "password");
+if (response.IsFailed) {
+    // Handle errors
+}
+```
+
+All subsequent operations via the logged in client will send along the stored access token.
+
+If your server additionally has namespaces (Dgraph v21.03 or above), use `LoginIntoNamespace`.
+
+```c#
+var response = await dgraphClient.LoginIntoNamespace("user", "password", 0x10);
+if (response.IsFailed) {
+    // Handle errors
+}
+```
+
+### Connecting To Dgraph Cloud
+
+Use `DgraphCloudChannel.Create` to create a GrpcChannel that connects to a Dgraph Cloud backend.
```

Comment:
Does this PR improve the code and takes all kinds of URLs like in these PRs:
1. https://github.com/dgraph-io/dgo/pull/159
2. https://github.com/dgraph-io/dgo/pull/157